pkg/lifecycle/daemon/statusonly/daemon.go

```diff
@@ -34,10 +34,7 @@ func (d *StatusReceiver) ClearProgress() {
 
 func (d *StatusReceiver) PushStreamStep(ctx context.Context, messages <-chan daemontypes.Message) {
 	debug := level.Debug(log.With(d.Logger, "method", "pushStreamStep"))
-	select {
-	case <-ctx.Done():
-		debug.Log("event", "ctx.Done", "err", ctx.Err())
-	case msg := <-messages:
+	for msg := range messages {
```

Comment:
not immediately obvious here but we only need to listen to messages on this channel until it's closed, which happens before the context is finished

Reply:
I think we might lose some cancellation logic here?

Reply:
discussed irl - cancellation will be taken care of by executor